native/src/binaryRead.h

```diff
@@ -544,7 +544,9 @@ struct RouteDataObject {
 				// priority over default
 				maxSpeed = r.maxSpeed(RouteTypeRule::PROFILE_NONE) > 0 ? r.maxSpeed(RouteTypeRule::PROFILE_NONE) : maxSpeed;
 				maxProfileSpeed = r.maxSpeed(profile) > 0 ? r.maxSpeed(profile) : maxProfileSpeed;
-			}
+            } else if (r.maxSpeed(RouteTypeRule::PROFILE_NONE) > 0) {
+                maxSpeed = r.maxSpeed(RouteTypeRule::PROFILE_NONE);
```

Comment:
@nnngrach Why we ignore maxspeed:hgv ( type = MAXSPEED + PROFILE_TRUCK ) ?

@vshcherb according this [code](https://github.com/osmandapp/OsmAnd/blob/master/OsmAnd-java/src/main/java/net/osmand/binary/BinaryMapRouteReaderAdapter.java#L276) we ignore maxspeed:forward and maxspeed:backward when detect maxspeed. I think need to fix this code to store maxspeed and change getMaximumSpeed method according to three state of [forward value](https://github.com/osmandapp/OsmAnd/blob/e44bc4f2b077c15a2d44ed9cb7a40561904cd0d1/OsmAnd-java/src/main/java/net/osmand/binary/BinaryMapRouteReaderAdapter.java#L274) backward (-1) both (0) forward (0)

Reply:
@ivanPyrohivskyi  I made some changes. Is this correct?